### PR TITLE
api: remove unused fields and methods

### PR DESCRIPTION
**Pull request by aaithal**
### Summary

api: remove unused fields and methods

### Implementation details

1. Removed Container.SentStatus
2. Removed Task.UpdateKnownStatusAndTime as it's functionally equivalent to Task.SetKnownStatus
3. Moved TaskStatus methods from containerstatus to taskstatus

### Testing

- [X] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [X] Unit tests on Linux (`make test`) pass
- [X] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: No, this is mostly getting rid of some unused code

### Description for the changelog

None

### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: agent/api/task.go [hunk 1]
```diff
@@ -66,32 +66,39 @@ type Task struct {
...
+	// and `GetKnownStatus`.
+	// TODO KnownStatus should probably be private with appropriately written
+	// setter/getter.  When this is done, we need to ensure that the UnmarshalJSON
+	// is handled properly so that the state storage continues to work.
 	KnownStatus     TaskStatus
 	knownStatusLock sync.RWMutex
-	// KnownStatusTime captures the time when the KnownStatus was last updated.  Do not access KnownStatusTime directly,
-	// instead use `GetKnownStatusTime`.
+	// KnownStatusTime captures the time when the KnownStatus was last updated.
+	// NOTE: Do not access KnownStatusTime directly, instead use `GetKnownStatusTime`.
```
Comment (richardpen) [inline agent/api/task.go]: Can this be private?

Comment (samuelkarp) [inline agent/api/task.go]: No, because then `json.Marshal` won't be able to see it.

Comment (samuelkarp) [inline agent/api/task.go]: Though thinking about this more, we could rename `KnownStatusTime` (and others) to `KnownStatusTimeUnsafe` and still let it marshal with the `json:"KnownTime"` tag.

Comment (aaithal) [inline agent/api/task.go]: Let's do the Unsafe thing. I'll work on that change in the scope of this PR.

File: agent/api/container.go [hunk 2]
```diff
@@ -62,47 +62,54 @@ type Container struct {
...
-	// use `GetDesiredStatus` and `SetDesiredStatus`.
-	// TODO DesiredStatus should probably be private with appropriately written setter/getter.  When this is done, we need
-	// to ensure that the UnmarshalJSON is handled properly so that the state storage continues to work.
+	// DesiredStatus represents the state where the container should go. Generally,
+	// the desired status is informed by the ECS backend as a result of either
+	// API calls made to ECS or decisions made by the ECS service scheduler,
+	// though the agent may also set the DesiredStatus if a different "essential"
+	// container in the task exits. The DesiredStatus is almost always either
+	// ContainerRunning or ContainerStopped.
+	// NOTE: Do not access DesiredStatus directly.  Instead, use `GetDesiredStatus`
```
Comment (samuelkarp) [inline agent/api/container.go]: Can you change `DesiredStatus`, `KnownStatus`, and `SentStatus` to have `Unsafe` as well?  These have the same caveats about not accessing them without the lock.

File: agent/acs/handler/payload_handler_test.go [hunk 3]
```diff
@@ -370,16 +370,16 @@ func TestAddPayloadTaskAddsNonStoppedTasksAfterStoppedTasks(t *testing.T) {
 	if firstTaskAdded.Arn != stoppedTaskArn {
 		t.Errorf("Expected first task arn: %s, got: %s", stoppedTaskArn, firstTaskAdded.Arn)
 	}
-	if firstTaskAdded.DesiredStatus != api.TaskStopped {
-		t.Errorf("Expected first task state be be: %s , got: %s", "STOPPED", firstTaskAdded.DesiredStatus.String())
+	if firstTaskAdded.DesiredStatusUnsafe != api.TaskStopped {
```
Comment (samuelkarp) [inline agent/acs/handler/payload_handler_test.go]: This really points out how our test code is doing things different from production code.  This should probably call `GetDesiredStatus` instead.

File: agent/api/container.go (hunk 2 above)
Comment (aaithal) [inline agent/api/container.go]: I haven't gotten to `containers.go` yet. Working my way through `task.go`. Will change these when I get there. I'll add a comment for when I think this PR is ready to be reviewed again.

**Comment by aaithal** [comment]
@samuelkarp @richardpen I have added more commits for suffixing states with `Unsafe`. Thankfully, we've caught a couple of potential race conditions as a result :)

Can you please review this PR again? Thanks!

**Pull request merged**